indexing-service/src/main/java/org/apache/druid/indexing/common/TaskStorageDirTracker.java

```diff
@@ -145,6 +149,7 @@ public synchronized StorageSlot pickStorageSlot(String taskId)
       final StorageSlot candidateSlot = slots[currIncrement % slots.length];
       if (candidateSlot.runningTaskId == null) {
         candidateSlot.runningTaskId = taskId;
+        ++numUsedSlots;
```

Comment:
For symmetry with the decrement (even though the operation already happens inside a lock), this should happen before `candidateSlot.runningTaskId` is assigned. Same comment for the other place in the code assigning this variable.